src/haddock/modules/sampling/rigidbody/__init__.py

```diff
@@ -148,7 +153,9 @@ def run(self, **params):
 
                 # Create a model for the expected output
                 model = PDBFile(output_pdb_fname, path=self.path)
-                model.topology = [e.topology for e in combination]
+                # FIXME: Investigate this topology structure and simplify
+                topology_list = [e.topology for e in combination]
+                model.topology = list(chain(*topology_list))
```

Comment:
This is redundant. `model.topology = topology_list` is enough.

Reply:
`topology_list` might be a nested list